cmd/webhook.go

```diff
@@ -174,13 +174,10 @@ func addVolume(target, added []corev1.Volume, basePath string) (patch []patchOpe
 func updateAnnotation(target map[string]string, added map[string]string) (patch []patchOperation) {
 	for key, value := range added {
 		if target == nil || target[key] == "" {
-			target = map[string]string{}
 			patch = append(patch, patchOperation{
 				Op:   "add",
-				Path: "/metadata/annotations",
-				Value: map[string]string{
-					key: value,
-				},
+				Path: "/metadata/annotations/" + key,
```

Comment:
do we need to handle the special character in annotation key?

Reply:
otherwise it report the following error:
```
Error from server (InternalError): Internal error occurred: add operation does not apply: doc is missing path: "/metadata/annotations/sidecar-injector-webhook.morven.me/status": missing value
```

Reply:
If you need to refer to a key with ~ or / in its name, you must escape the characters with \~0 and \~1 respectively. For example, to get "baz" from { "foo/bar~": "baz" } you’d use the pointer /foo\~1bar\~0.

http://jsonpatch.com/